src/bitdrift_public/protobuf/workflow/v1/workflow.proto

```diff
@@ -227,6 +228,68 @@ message Workflow {
       message Histogram {
       }
     }
+
+    // Emit data for Sankey diagram.
+    message ActionSankeyDiagram {
+      // The ID of a Sankey diagram. If multiple diagrams have them same ID and are performed
+      // as the result of processing the same event (i.e. log) a client performs only one
+      // of the actions. The client emit a stat with a name that follows
+      // the following format "workflow_actions::<ID>".
+      // As the identifier becomes a part of metric's name it must meet the requirements
+      // of prometheus metric name. In other words it needs to match
+      // regex "[a-zA-Z_][a-zA-Z0-9_]*".
+      string id = 1 [(validate.rules).string = {min_len: 1}];
+
+      // A state step provides the configuration used to name Sankey chart states. Each state step corresponds
+      // to a single transition between two states in the workflow.
+      message NodeStep {
```

Comment:
For associating with the transition I had thought we could just embed it directly on the Transition proto / LogMatcher proto, that way we avoid having to duplicate all the state transitions and it ought to make it easier to construct the necessary bits on the workflow engine

That said I haven't spent time looking at the client code so lmk if you think this format works better